### ISSUE TITLE
mousepad not working with mutter-3.36.3

### ISSUE THREAD

**Issue by frosth555** [andyholmes/gnome-shell-extension-gsconnect#869]
Archlinux - Gnome 3.36 with gnome-shell/mutter update to 3.36.3

After update, mousepad stopped working, I can see share-desktop icon in panel, see communication in logs but cursor don't move, no tap, no swipe... 
Downgrade mutter to previously version (3.36.2) bring back mousepad to live.  

log:
```
GSConnect Version: 38
GSConnect Install: user
GJS: 16403
XDG_SESSION_TYPE: wayland
GDMSESSION: gnome
--------------------------------------------------------------------------------
-- Logs begin at Fri 2020-05-29 08:42:23 CEST, end at Fri 2020-06-05 16:17:41 CEST. --
cze 05 16:17:36 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/protocol/lan.js:_onIdentity:315]: {
                                                              "id": 1591366656739,
                                                              "type": "kdeconnect.identity",
                                                              "body": {
                                                                "deviceId": "5f97eed13b44c9c0",
                                                                "deviceName": "Redmi 4",
                                                                "protocolVersion": 7,
                                                                "deviceType": "phone",
                                                                "incomingCapabilities": [
                                                                  "kdeconnect.telephony.request_mute",
                                                                  "kdeconnect.notification",
                                                                  "kdeconnect.ping",
                                                                  "kdeconnect.notification.reply",
                                                                  "kdeconnect.notification.action",
                                                                  "kdeconnect.share.request",
                                                                  "kdeconnect.clipboard.connect",
                                                                  "kdeconnect.runcommand",
                                                                  "kdeconnect.contacts.request_all_uids_timestamps",
                                                                  "kdeconnect.sms.request_conversations",
                                                                  "kdeconnect.telephony.request",
                                                                  "kdeconnect.mpris",
                                                                  "kdeconnect.sms.request_conversation",
                                                                  "kdeconnect.findmyphone.request",
                                                                  "kdeconnect.systemvolume",
                                                                  "kdeconnect.mousepad.keyboardstate",
                                                                  "kdeconnect.sftp.request",
                                                                  "kdeconnect.share.request.update",
                                                                  "kdeconnect.notification.request",
                                                                  "kdeconnect.mousepad.request",
                                                                  "kdeconnect.photo.request",
                                                                  "kdeconnect.sms.request",
                                                                  "kdeconnect.contacts.request_vcards_by_uid",
                                                                  "kdeconnect.mpris.request",
                                                                  "kdeconnect.battery.request",
                                                                  "kdeconnect.clipboard"
                                                                ],
                                                                "outgoingCapabilities": [
                                                                  "kdeconnect.sms.messages",
                                                                  "kdeconnect.telephony",
                                                                  "kdeconnect.mpris",
                                                                  "kdeconnect.notification",
                                                                  "kdeconnect.contacts.response_uids_timestamps",
                                                                  "kdeconnect.findmyphone.request",
                                                                  "kdeconnect.ping",
                                                                  "kdeconnect.mousepad.keyboardstate",
                                                                  "kdeconnect.share.request",
                                                                  "kdeconnect.contacts.response_vcards",
                                                                  "kdeconnect.notification.request",
                                                                  "kdeconnect.mousepad.echo",
                                                                  "kdeconnect.clipboard.connect",
                                                                  "kdeconnect.mousepad.request",
                                                                  "kdeconnect.presenter",
                                                                  "kdeconnect.sftp",
                                                                  "kdeconnect.photo",
                                                                  "kdeconnect.runcommand.request",
                                                                  "kdeconnect.mpris.request",
                                                                  "kdeconnect.systemvolume.request",
                                                                  "kdeconnect.battery",
                                                                  "kdeconnect.clipboard"
                                                                ],
                                                                "tcpPort": whatever,
                                                                "tcpHost": "192.168.44.23"
                                                              }
                                                            }
cze 05 16:17:37 systemd[1]: systemd-hostnamed.service: Succeeded.
cze 05 16:17:37 audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-hostnamed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
cze 05 16:17:37 audit: BPF prog-id=21 op=UNLOAD
cze 05 16:17:37 audit: BPF prog-id=20 op=UNLOAD
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/protocol/core.js:receive/<:298]: Redmi 4: {
                                                              "id": 1591366659676,
                                                              "type": "kdeconnect.mousepad.request",
                                                              "body": {
                                                                "dx": 0,
                                                                "dy": 0
                                                              }
                                                            }
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/protocol/core.js:receive/<:298]: Redmi 4: {
                                                              "id": 1591366659693,
                                                              "type": "kdeconnect.mousepad.request",
                                                              "body": {
                                                                "dx": 0,
                                                                "dy": -3
                                                              }
                                                            }
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/components/input.js:_ensureAdapter:414]: "Creating Mutter RemoteDesktop session"
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/components/input.js:movePointer:455]: this._session is null
                                                            movePointer@/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/components/input.js:455:13
                                                            _handleInput@/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:171:29
                                                            handlePacket@/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/plugins/mousepad.js:145:26
                                                            receivePacket@/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/device.js:441:29
                                                            receive/<@/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/protocol/core.js:299:32
                                                            @/home/arek/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js:1133:17
cze 05 16:17:39 gnome-shell[827]: g_udev_client_query_by_device_file: assertion 'device_file != NULL' failed
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/protocol/core.js:receive/<:298]: Redmi 4: {
                                                              "id": 1591366659709,
                                                              "type": "kdeconnect.mousepad.request",
                                                              "body": {
                                                                "dx": 1,
                                                                "dy": -8
                                                              }
                                                            }
cze 05 16:17:39 org.gnome.Shell.Extensions.GSConnect[1282]: [/service/protocol/core.js:receive/<:298]: Redmi 4: {
                                                              "id": 1591366659726,
                                                              "type": "kdeconnect.mousepad.request",
                                                              "body": {
                                                                "dx": -2,
                                                                "dy": -14
                                                              }
                                                            }


```
edit:trim log a little bit

**Comment by andyholmes** [comment]
It doesn't seem like anything is going wrong on our side. Is there anything in your regular logs indicating error? Is it possible you disabled the remote desktop functionality ofr some reason?

**Comment by frosth555** [comment]
Nope, as I said remote-desktop indicator is shown on panel. Installing previous mutter version and then relogin works without any other actions.
In journal...I didn't notice it yesterday,  it appears always when trying use mousepad:
`gnome-shell[799]: g_udev_client_query_by_device_file: assertion 'device_file != NULL' failed`

I'll try it on fedora to eliminate distro bug, next bisecting mutter if needed.

**Comment by smarnv** [comment]
I experienced the same issue on the same version of Gnome (3.36) and mutter (3.36.3), and have the same stack trace. I can also post it, but the header with GSConnect version/install, XDG session type, etc is also the same, so it seems like I was able to reproduce it exactly.

If I understood the code correctly, it seems that the promise in [src/service/components/input.js#L363](../blob/b025ea88567c26b995b2a89a14ba821d90f9bc85/src/service/components/input.js#L363) doesn't return the session when it's expected for the input as mousepad. The [aforementioned commit](https://gitlab.gnome.org/GNOME/mutter/-/commit/3daf9124493ba556b68ccff12dfb9e282661341b) changes how mutter checks for permission for remote desktop input, but unfortunately I couldn't find out how exactly it affects GSConnect so far, and how it can be fixed. :man_shrugging:
